### PR TITLE
osa: add daily job that tests production deployments

### PR DESCRIPTION
**Pull request by 0xmichalis**
@openshift/sig-azure this PR adds a daily job in CI that runs cluster deployments via the e2e RP framework using the production RP.

### CODE REVIEW

File: ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml [hunk 1]
```diff
@@ -218,3 +218,54 @@ periodics:
...
+  cron: ""
+  decorate: true
+  interval: 24h
+  name: azure-test-production-resource-provider
+  skip_cloning: true
+  spec:
+    containers:
+    - args:
+      - --secret-dir=/usr/local/e2e-secrets/azure
+      - --target=[output:stable:sync]
```
Comment (kargakis) [inline ci-operator/jobs/openshift/openshift-azure/openshift-openshift-azure-periodics.yaml]: This is not really needed; I will see whether we can get rid of it w/o needing any changes in the template in a follow-up.

**Comment by kargakis** [comment]
We are using master to run the e2e tests atm. We probably want to match with the commit that is run in prod (so I guess we should be using the private preview branch?).

**Comment by kwoodson** [comment]
@kargakis 
> We are using master to run the e2e tests atm. We probably want to match with the commit that is run in prod (so I guess we should be using the private preview branch?).

Using a commit ID, tag, or branch makes sense to me.

**Comment by kargakis** [comment]
Unfortunately we cannot use the private preview branch because we would have to backport all the e2e framework machinery to it. For now, we should be fine with master I think. Will try this out and see how it works.

**Pull request merged**